File: php/src/Model/SupervisorCounts.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * The board's headline numbers.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class SupervisorCounts
{
    public function __construct(
        public readonly int $queued,
        public readonly int $pending,
        public readonly int $parked,
        public readonly int $oldestWaitMs,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            queued: (int) ($data['queued'] ?? 0),
            pending: (int) ($data['pending'] ?? 0),
            parked: (int) ($data['parked'] ?? 0),
            oldestWaitMs: (int) ($data['oldestWaitMs'] ?? 0),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/SupervisorCounts.java
package io.fivexer.sdk.model;

/**
 * The board's headline numbers.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class SupervisorCounts {
    private Integer queued;
    private Integer pending;
    private Integer parked;
    private Long oldestWaitMs;

    public Integer getQueued() { return queued; }
    public Integer getPending() { return pending; }
    public Integer getParked() { return parked; }
    public Long getOldestWaitMs() { return oldestWaitMs; }
}
